Userland/Applications/Screenshot/ScreenshotWidget.cpp

```diff
@@ -0,0 +1,155 @@
+    m_copy_checkbox = *find_descendant_of_type_named<GUI::CheckBox>("copy_to_clipboard");
+    m_cancel_button = *find_descendant_of_type_named<GUI::Button>("cancel_button");
+    m_ok_button = *find_descendant_of_type_named<GUI::Button>("ok_button");
+
+    m_window_sel = GUI::Window::construct();
+    m_overlay = m_window_sel->set_main_widget<SelectableOverlay>(m_window_sel);
+    m_overlay->callback = [&] {
+        save_screenshot(true);
+    };
+
+    m_window_sel->set_title("Screenshot");
+    m_window_sel->set_has_alpha_channel(true);
+    m_window_sel->set_fullscreen(true);
+
+    String type = Config::read_string("Screenshot", "General", "ScreenshotType", "Whole");
+    if (type == "Custom") {
+        m_whole_button->set_checked(false);
+        m_custom_button->set_checked(true);
+    } else {
+        m_whole_button->set_checked(true);
+        m_custom_button->set_checked(false);
+    }
+
+    m_delay_spinbox->set_value(Config::read_i32("Screenshot", "General", "Delay", 0));
+    m_copy_checkbox->set_checked(Config::read_bool("Screenshot", "General", "CopyToClipboard", true));
+
+    m_cancel_button->on_click = [&](auto) {
+        GUI::Application::the()->quit();
+    };
+
+    m_ok_button->on_click = [&](auto) {
+        String new_type = "Whole";
+        if (m_custom_button->is_checked()) {
+            new_type = "Custom";
+        }
+
+        Config::write_string("Screenshot", "General", "ScreenshotType", new_type);
+        Config::write_i32("Screenshot", "General", "Delay", m_delay_spinbox->value());
+        Config::write_bool("Screenshot", "General", "CopyToClipboard", m_copy_checkbox->is_checked());
+
+        m_output_path = Core::DateTime::now().to_string("screenshot-%Y-%m-%d-%H-%M-%S.png");
+
+        window()->set_frameless(true);
+        window()->set_opacity(0);
+        start_timer(m_delay_spinbox->value());
+    };
+}
+
+void ScreenshotWidget::save_screenshot(bool from_callback)
+{
+    Gfx::IntRect crop_region;
+    if (m_custom_button->is_checked()) {
+        if (from_callback) {
+            crop_region = m_overlay->region();
+            if (crop_region.is_empty()) {
+                return;
+            }
+        } else {
+            m_window_sel->show();
+            return;
+        }
+    }
+
+    dbgln("Taking screenshot");
+    Optional<u32> screen_index;
+    auto screen_bitmap = GUI::WindowServerConnection::the().get_screen_bitmap(crop_region, screen_index);
+    dbgln("Taken screenshot");
+
+    RefPtr<Gfx::Bitmap> bitmap = screen_bitmap.bitmap();
+    if (!bitmap) {
+        warnln("Failed to grab screenshot!");
+        return;
+    }
+
+    if (m_copy_checkbox->is_checked()) {
+        GUI::Clipboard::the().set_bitmap(*bitmap);
+
+        auto notification = GUI::Notification::construct();
+        notification->set_title("Screenshot taken");
+        notification->set_text("Screenshot saved to clipboard");
+        notification->show();
```

Comment:
An icon to an notification would be nice ;)

```cc
        notification->set_icon(GUI::Icon::default_icon("app-screenshot").bitmap_for_size(32));
```
![proposed preview](https://user-images.githubusercontent.com/16520278/131513332-d83b368f-2f8b-47af-81a3-54c17bf5533a.png)